### PR TITLE
[MM-58179] Review networking performance at scale

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

As you know, this turned out to be slightly more extensive than anticipated. Please refer to the [investigation](https://community.mattermost.com/core/pl/csdfuynxbbyiukt3rwwjq1hi7c) for the rationale behind the proposed changes. 

One thing I haven't done yet is attempt a 100k load test. All my attempts reached 20k since that was sufficient to trigger the symptoms. Let me know if you think there's value in doing so while I have the setup on. 

There's a decent chance doing so would cause the coordinator to hit the 1% threshold, at which point my recommendation would be to vertically scale the proxy instance further, even if overall underutilized, rather than accepting it and blindly bumping the threshold. 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-58179

### CODE REVIEW

File: deployment/terraform/strings.go
Comment (streamer45) [inline deployment/terraform/strings.go:302]: These were not needed since the second value in `tcp_rmem` and `tcp_wmem` overrides them.

**Comment by agnivade** [comment]
> There's a decent chance doing so would cause the coordinator to hit the 1% threshold,

Can you clarify more on how you reached the 1% number? Was 1% the rate without the optimization when you were trying to reach 20K?

There's no need for a full blown 100K test. I'd like to see what the rate is for a 50K test (with 100M data set) without the optimization and extrapolate that to a 100K level and set that as the threshold. Basically, my thinking is that since we were able to use the app at 100K without any TCP thresholds, now with the optimization, the new thresholds shouldn't hit.

**Comment by streamer45** [comment]
> > There's a decent chance doing so would cause the coordinator to hit the 1% threshold,
> 
> Can you clarify more on how you reached the 1% number? Was 1% the rate without the optimization when you were trying to reach 20K?

The 1% value is arbitrary, but it's a generally accepted threshold since it's indirectly tracking packet loss, which in our local environment (in my opinion) shouldn't be anywhere near that on average. 

> There's no need for a full blown 100K test. I'd like to see what the rate is for a 50K test (with 100M data set) without the optimization and extrapolate that to a 100K level and set that as the threshold. Basically, my thinking is that since we were able to use the app at 100K without any TCP thresholds, now with the optimization, the new thresholds shouldn't hit.

That would be great. Another set of eyes may be good.

File: deployment/terraform/strings.go
Comment (streamer45) [reply deployment/terraform/strings.go:297]: Yes, it should be 4x. The comment is wrong and not the value tested.

File: deployment/terraform/create.go
Comment (streamer45) [reply deployment/terraform/create.go:754]: Adding a comment is the easiest. We can also document it separately, but then it's two places to keep up to date. And, of course, we may need to surface this in our documentation eventually if customers want to follow our recommendations.

Comment (streamer45) [reply deployment/terraform/create.go]: @agarciamontoro I added a comment for now and made it dynamic since the higher value (8192) may not work for all instance types. We may surface it in config later if we need an extra level of tuning.

Comment (agarciamontoro) [reply deployment/terraform/create.go]: Neat, thanks!

**Review by agnivade — APPROVED** [review]
Had a chat with @agarciamontoro on this. He wants to start his ceiling tests v2 with this included. There's really no need to block this on my work to fine tune the thresholds. The thresholds can be tuned later.

**Pull request merged**
